src/app/detail/[task_id]/[user_type]/DetailStateUpdate.tsx

```diff
@@ -0,0 +1,39 @@
+import { ClientSideStateUpdate } from '@/hoc/ClientSideStateUpdate'
+import { getAllTasks, getAllWorkflowStates, getViewSettings } from '@/app/page'
+import { Token } from '@/types/common'
+
+interface DetailStateUpdateProps {
+  isRedirect?: 'true'
```

Comment:
any specific reason to use string `true` @rrojan

Reply:
Switched to use `&isRedirect` instead of `&isRedirect=true`

Reply:
I used to use `query-string` from npm to parse the search param but looks like that won't work here as well, `query-string` accepts raw `searchParam`.

However, my actual concerned is  component `DetailStateUpdate` defining prop as `isRedirect?: 'true'`. this couples the component to parent component, rather parent component could add the logic to send `boolean`, search param issue should not be issue of `DetailStateUpdate` component IMO